src/sonic-config-engine/minigraph.py

```diff
@@ -1293,6 +1292,9 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
         }
     }
 
+    if deployment_id:
```

Comment:
>deployment_id [](http://example.com/codeflow?start=7&length=13)

It is working with current code. I am worried about if anyone will accidently modify the implementation of `parse_meta()`, changing deployment_id from str to int. And if the deployment_id == 0, it will be skipped in `result`.

Suggest check `deployment_id is not None`.